### PR TITLE
feat: add subTags UI (GSoC)

### LINKED ISSUE #2195: ADMIN: Implement SubTags UI (GSoC)
### Describe the feature

Adding support for `subTags` in the admin.

### Describe the solution you'd like

Making the SubTags UI, adding options to go to the subTags of a given tag.
subTags screen

### Additional context

Previous Issue: #2184

### PR DESCRIPTION
**Pull request by meetulr**
### What kind of change does this PR introduce?

Implements SubTags UI.

### Issue Number:

Fixes #2195 

### Did you add tests for your changes?

Yes

### Does this PR introduce a breaking change?

No

### Snapshots/videos

https://github.com/user-attachments/assets/6d2ab631-3cee-4de8-b47d-7e366402bce9

### Additional Context

Previous PR: #2185

## Summary by CodeRabbit

- **New Features**
	- Introduced a new route for managing sub-tags, enhancing navigation within the application.
	- Added a comprehensive `SubTags` component for viewing, adding, and removing sub-tags.
	- Implemented pagination for sub-tags display.
	- Enhanced error handling and user feedback through notifications for tag operations.

- **Styling Improvements**
	- New CSS classes for improved styling of error messages and tag UI components.
	- Added responsive design elements to ensure a better user experience across devices.

- **Bug Fixes**
	- Updated routing paths for clarity and consistency in navigation within the application.

### CODE REVIEW

**Comment by meetulr** [comment]
@coderabbitai review please

File: src/screens/ManageTag/ManageTag.tsx [hunk 1]
```diff
@@ -166,10 +166,10 @@ function ManageTag(): JSX.Element {
   const orgUserTagAncestors = orgUserTagAncestorsData?.getUserTagAncestors;
 
   const goToSubTags = (tagId: string): void => {
-    navigate(`/orgtags/${currentUrl}/orgtagSubTags/${tagId}`);
+    navigate(`/orgtags/${currentUrl}/subTags/${tagId}`);
   };
 
-  const handleClick = (tagId: string): void => {
+  const goToManageTag = (tagId: string): void => {
```
Comment (beingnoble03) [inline src/screens/ManageTag/ManageTag.tsx:172]: how about changing`goToManageTag ` to `redirectToManageTag`?

File: src/screens/ManageTag/ManageTag.tsx [hunk 2]
```diff
@@ -166,10 +166,10 @@ function ManageTag(): JSX.Element {
...
     navigate(`/orgtags/${currentUrl}/managetag/${tagId}`);
```
Comment (beingnoble03) [inline src/screens/ManageTag/ManageTag.tsx:173]: `currentUrl` needs to be changed to `organizationID`? It's a bit confusing here. I think I missed this in the last PR.

File: src/screens/SubTags/SubTags.tsx [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+  '& .MuiDataGrid-row.Mui-hovered': {
+    backgroundColor: 'transparent',
+  },
+  '& .MuiDataGrid-root': {
+    borderRadius: '0.1rem',
+  },
+  '& .MuiDataGrid-main': {
+    borderRadius: '0.1rem',
+  },
+};
```
Comment (beingnoble03) [inline src/screens/SubTags/SubTags.tsx:49]: Looks like it's repeated. Can we declare this in a separate common style file?

File: src/screens/SubTags/SubTags.tsx [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+        toast.success(t('tagCreationSuccess'));
+        subTagsRefetch();
+        setTagName('');
+        setAddSubTagModalIsOpen(false);
+      }
+    } catch (error: unknown) {
+      /* istanbul ignore next */
+      if (error instanceof Error) {
+        toast.error(error.message);
+        console.log(error.message);
```
Comment (beingnoble03) [inline src/screens/SubTags/SubTags.tsx:151]: Please remove the debug statements, particularly the `console.log`s.

File: src/screens/SubTags/SubTags.tsx [hunk 5]
```diff
@@ -0,0 +1,587 @@
...
+
+function SubTags(): JSX.Element {
+  const { t } = useTranslation('translation', {
+    keyPrefix: 'organizationTags',
+  });
+  const { t: tCommon } = useTranslation('common');
+
+  const [addSubTagModalIsOpen, setAddSubTagModalIsOpen] = useState(false);
+
+  const { orgId: currentUrl, tagId: parentTagId } = useParams();
```
Comment (beingnoble03) [inline src/screens/SubTags/SubTags.tsx:59]: Same about this `currentUrl`, this needs to be changed to `orgId` or `organizationId`

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
@meetulr rest looks good. Please address the comments.

File: src/screens/ManageTag/ManageTag.tsx (hunk 2 above)
Comment (meetulr) [reply src/screens/ManageTag/ManageTag.tsx]: yes, that's right, thank you.

**Review by beingnoble03 — APPROVED** [review]
@palisadoes this needs one more approval. Would lean on other maintainers for another look.

**Review by palisadoes — APPROVED** [review]

**Review by Nitya-Pasrija — APPROVED** [review]

**Pull request merged**